### DISCUSSION TITLE
i need to call signalR hub methods in quartz IJob classes but it seems impossible.

### DISCUSSION THREAD

**Discussion by virtouso** [quartznet/quartznet#1718] [General]
Hi. 

first of all is it possible to use quartz without IJob classes and schedule a job just directly by methods?

to be able to call signalr methods on schedule i need to inject the hub on IJob but examples on the net doesn't work. 

is there any way around for it to make it work?

**Comment by lahma** [comment]
Maybe you could create a full repro to demonstrate the problem? Otherwise it's all guessing game.

**Comment by virtouso** [comment]
this is something that i try to do:```


```c#
    public class FinishGameJob : Quartz.IJob
    {
        private readonly IServiceProvider _serviceProvider;


        public FinishGameJob(IServiceProvider serviceProvider)
        {
            _serviceProvider = serviceProvider;
        }

        public Task Execute(IJobExecutionContext context)
        {
            JobDataMap dataMap = context.JobDetail.JobDataMap;
            string matchId = dataMap.GetString("matchId");
          
            EndGameReason reason = (EndGameReason)dataMap.GetInt("reason");
            using (var scope = _serviceProvider.CreateScope())
            {
                var hc = scope.ServiceProvider.GetRequiredService<IHubContext<GuessGamePausableMatchHub>>();
                 hc.Clients.All.SendAsync("Notify", "Wub-laba-dub-da");
            }
            
            
            return Task.CompletedTask;
        }
    }
```

but not working

**Reply by lahma** [reply]
And the error is?

**Reply by virtouso** [reply]
in general, using quartz and signalr together is something many people have problem with it in the net and there is no solution. if you are a contributor please make an official solution for that. that will help lots of people.

**Reply by lahma** [reply]
Maybe you can create a PR for that we'll see if it gets merged and available to all?

**Comment by virtouso** [comment]
it never gets scheduled. no bug. as its said i have to manually inject the dependencies. but dont know how to use signalr for it. its the sample i use for it:

`      var services = new ServiceCollection();
          
            var container = services.BuildServiceProvider();
            var jobFactory = new JobFactory(container);
            services.AddScoped<IHost>();
            services.AddScoped<IHubContext<GuessGamePausableMatchHub>>();
            scheduler.JobFactory = jobFactory;
         `

**Comment by virtouso** [comment]
in general, is it possible to just schedule a method like the hangfire and avoid using IJob?

**Reply by lahma** [reply]
There's no support for delegate invocations out of the box, if you need something like that maybe hangfire is the solution for you?
